frb_codegen/src/config.rs

```diff
@@ -187,10 +191,57 @@ pub fn parse(raw: RawOpts) -> Vec<Opts> {
                 .collect::<Vec<_>>()
         })
         .unwrap_or_else(|| {
-            vec![fallback_c_output_path()
-                .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("c_output")))]
+            (0..rust_input_paths.len())
+                .map(|_| {
+                    fallback_c_output_path()
+                        .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("c_output")))
+                })
+                .collect()
         });
 
+    if c_output_paths.len() != rust_input_paths.len() {
+        bail(
+            clap::ErrorKind::WrongNumberOfValues,
+            "--c-output's inputs should match --rust-input's length".into(),
+        );
+    }
+    // 2.extra c path from flag rawOpt.extra_c_output_path
+    let extra_c_output_paths = raw
+        .extra_c_output_path
+        .unwrap_or_default()
+        .iter()
+        .flat_map(|extra_path| {
+            c_output_paths.iter().map(move |c| {
+                let path = Path::new(c);
+                let file_name = String::from(path.file_name().unwrap().to_str().unwrap());
+                Path::join(extra_path.as_ref(), file_name)
+                    .into_os_string()
+                    .into_string()
+                    .unwrap()
+            })
+        })
+        .collect::<Vec<_>>();
+
+    // 3.integrate c output path(s) for each rust input API block
+    let refined_c_outputs = c_output_paths
+        .iter()
+        .enumerate()
+        .map(|(i, each_a)| {
+            let mut first = vec![each_a.clone()];
+            if !extra_c_output_paths.is_empty() {
+                let iter = extra_c_output_paths[i..]
+                    .iter()
+                    .step_by(c_output_paths.len())
+                    .map(|s| s.to_owned())
+                    .collect::<Vec<_>>();
+                first.extend(iter);
+            }
+
+            first
+        })
+        .collect::<Vec<_>>();
+    //↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑ c output path(s) ↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑
+
```

Comment:
would be great to extract to a separate function if possible

Reply:
It is moved into a new function called `get_refined_c_output`, cd09bcb737e03e8875b477db2a0a474837bc7c7e.